lib/acme/client/resources/authorization.rb

```diff
@@ -7,12 +7,26 @@ class Acme::Client::Resources::Authorization
 
   def initialize(client, response)
     @client = client
-    assign_challenges(response.body['challenges'])
+    @uri = response.headers['Location']
     assign_attributes(response.body)
   end
 
+  def verify_status
+    response = @client.connection.get(@uri)
+
+    assign_attributes(response.body)
+    status
+  end
```

Comment:
Mh, maybe. How about exposing `uri` and having a `Client#fetch_authorization(uri)`?